### PR TITLE
FINERACT-842 Made TRY-IT swagger button work

### PR DESCRIPTION
**Pull request by thesmallstar**
This PR can only be merged after: #1163

### CODE REVIEW

**Comment by thesmallstar** [comment]
After #1163 is merged we only need to add the following code to fineract-input.yaml ( can be converted from json to yaml automatically)

```
servers:
- url: /fineract-provider/api/v1
components:
  securitySchemes:
    basicAuth:
      type: http
      scheme: basic 
```

And the try it button would work, I have tested this :D

**Comment by vorburger** [comment]
@ptuomola do you want to review and locally test this one? I'm happy to just merge it if it has your LGTM... (or even let this be one of the PRs that you can merge yourself, soon).

**Comment by vorburger** [comment]
BTW I don't suppose there's an easy better way for that SecurityRequirement annotation to be specified only once (perhaps programmatically?), instead of having to be specified everywhere? It's not a huge deal, and I don't mind merging it as is, but was just thinking that people may forget to add that in new *Resource classes for new APIs in the future. Or perhaps they won't, because they'll just copy/paste anyway..  ;-)

**Comment by thesmallstar** [comment]
With my basic knowledge on this( and a lot of research :P ) I could find the only this way to work. 
I will search for a better option for this once more :D

**Comment by thesmallstar** [comment]
https://www.loom.com/share/8eb964074579491090db5bacbe5eff99

@vorburger @ptuomola  I have a small demo video on how this works :) 

The only problem I see is to give a proper error msg if the user try to access the API without authorizing.

**Comment by thesmallstar** [comment]
For the problem that if the user does not authorize, I think a better approach will be to have a the authorization parameters set by default to:
username: mifos
password: password

Since this would probably be used most (if not all) of the time for that username. 
This, however, is not part of this PR and I would raise a separate PR for that purpose.

**Comment by thesmallstar** [comment]
Just noticed, Some API routes are not covered yet, converting to draft to avoid accidental merge.

**Comment by thesmallstar** [comment]
@vorburger there was an easier (and probably the correct way) to do this. I have updated this. 
I have also added tenantid as an authorization param so that user can enter it if he wants.

By default, the user gets them set, and he can directly test the API :)

image


Ready for your review @ptuomola

**Review by vorburger — APPROVED** [review]
Oh! Wow. Ok, this is really cool... ;-) LGTM! I'd be (very) happy to merge this.. but haven't actually locally tested it myself - I'll let @ptuomola LGTM this.

**Comment by vorburger** [comment]
@thesmallstar wait but can the end-user actually CHANGE the uid/pwd (and - tenant) in the Swagg UI? That wasn't clear from the screenshot, to me.

**Comment by thesmallstar** [comment]
@vorburger yes, the initial values are only added for convenience since 90percent if the times we will probably need default id-pass and tenant.

File: fineract-provider/config/swagger/fineract-input.yaml
Comment (vorburger) [inline fineract-provider/config/swagger/fineract-input.yaml]: @thesmallstar could you remove this termsOfService as discussed in #1163 ?

**Comment by thesmallstar** [comment]
@ptuomola  WDYT?

**Comment by ptuomola** [comment]
@thesmallstar Looks great to me! Tried it out locally and it works fine

**Review by ptuomola — APPROVED** [review]

**Comment by thesmallstar** [comment]
@vorburger this is ready to be merged  😄

**Review by Sanyam96 — APPROVED** [review]

**Comment by Sanyam96** [comment]
LGTM 
Awesome work @thesmallstar

**Pull request merged**
